src/ios-deploy/ios-deploy.m

```diff
@@ -1463,6 +1500,10 @@ AFCConnectionRef start_house_arrest_service(AMDeviceRef device) {
 
 void list_files_callback(AFCConnectionRef conn, const char *name, read_dir_cb_reason reason)
 {
+    if (_json_output) {
```

Comment:
We should probably gate the call to read_dir() rather than return early inside it's callback.